keepassxc-browser/keepassxc-browser.js

```diff
@@ -825,51 +191,59 @@ cipFields.getAriaHidden = function(field) {
 };
 
 cipFields.getOverflowHidden = function(field) {
-    let $par = jQuery(field).parents();
-    for (const p of $par) {
-        const val = $(p).css('overflow');
-        if (val === 'hidden') {
+    let parents = [];
+    while (field.parentElement) {
+        parents.push(field = field.parentElement)
+    }
+
+    for (const p of parents) {
+        if (p.style.overflow === 'hidden') {
             return true;
         }
     }
     return false;

```

Comment:
`cipFields.getAriaHidden()` and `cipFields.getOverflowHidden()` can be refactored into:

```js
/**
 * Check aria-hidden attribute by looping the parent elements of input field
 * @param {HTMLElement} field
 * @returns {string}
 */
cipFields.getAriaHidden = function(field) {
  return traverseParents(field, val => val, val => val, () => "false");
};

/**
 * @param {HTMLElement} field
 * @returns {boolean}
 */
cipFields.getOverflowHidden = function(field) {
  return traverseParents(field, f => f.style.overflow === "hidden");
};

/**
 * Returns the first parent element satifying the {@code predicate} mapped by {@code resultFn} or else {@code defaultVal}.
 * @param {HTMLElement} element the start element (excluded, starting with the parents)
 * @param {function} predicate matcher for the element to find, type (HTMLElement) => boolean
 * @param {function} resultFn callback function of type (HTMLElement) => {*} called for the first matching element
 * @param {fun} defaultValFn fallback return value supplier, if no element matching the predicate can be found
 */
function traverseParents(
  element,
  predicate,
  resultFn = () => true,
  defaultValFn = () => false
) {
  for (let f = field.parentElement; f !== null; f = f.parentElement) {
    if (predicate(f)) {
      return resultFn(f);
    }
  }
  return defaultValFn();

```

Reply:
Good idea. Makes it more readable too.